File: app/src/main/java/net/daverix/ajvm/operation/ByteCodeOperation.java
package net.daverix.ajvm.operation;


import net.daverix.ajvm.Frame;
import net.daverix.ajvm.ByteCodeReader;

import java.io.IOException;

public interface ByteCodeOperation {
    void execute(ByteCodeReader reader,
                 int indexOfBytecode,
                 Frame currentFrame) throws IOException;
}

File: app/src/main/java/net/daverix/ajvm/operation/ByteCodeOperation.kt
package net.daverix.ajvm.operation


import net.daverix.ajvm.Frame
import net.daverix.ajvm.ByteCodeReader

import java.io.IOException

interface ByteCodeOperation {
    @Throws(IOException::class)
    fun execute(reader: ByteCodeReader,
                indexOfBytecode: Int,
                currentFrame: Frame)
}
